> any change to upstream the poetry scripts?

Yep, the fork adds them, they're not present at all in the main repository. There is an issue open which asks if upstream is interested in incorperating those changes, but it got ignored: https://github.com/jamesofarrell/i3-swallow/issues/25.

The poetry scripts are all included in 1 commit, so i tried to use the main repo and just fetch the patch that adds the poetry scripts, but I didn't have much luck:

```console
~/nix/nixpkgs > ./result/bin/swallow
Traceback (most recent call last):
  File "/nix/store/slqk69xr82vfp51aba825cbd5rirfc45-i3-swallow-1.0.0/bin/.swallow-wrapped", line 6, in <module>
    from i3_swallow import main
  File "/nix/store/slqk69xr82vfp51aba825cbd5rirfc45-i3-swallow-1.0.0/lib/python3.9/site-packages/i3_swallow/__init__.py", line 1, in <module>
    from i3_swallow.swallow import main
ModuleNotFoundError: No module named 'i3_swallow.swallow'
```
Relevant lines from the diff:
```diff
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-ROHHp6yx7rjkA6G2+jULW6Vv5gh4CXy1WHPM/PVpdrg=";
+  src = fetchFromGitHub {
+    owner = "jamesofarrell";
+    repo = "i3-swallow";
+    rev = "7a9faed693b0da7df762a095fed512eca8c6414c";
+    sha256 = "1fq9cvisimgyza8c91bkill9wgyj7ac9sf9yz87npc1smlgscibd";
   };


+  patches = [ (fetchpatch {
+    url = "https://github.com/ViliamV/i3-swallow/commit/efd7af186c024ed5bf6036fc6cdb4762f08b7c55.patch";
+    sha256 = "sha256-CVW3aWAHmuQUn+FXT0UpYkXaneMPcWFwBXAYjqmaTxc=";
+  }) ];
```

I really don't understand how this doesn't work, since that commit contains the scripts in the same state as on the pip package, they have not changed at all. Only other change from the forked repo is some readme stuff, which we don't really care about.

>  also now that one repo has a poetry definition can't we use poetry2nix instead of a manual derivation ?

I'm not extremely familair with python packaging (also probably the reason I can't get the main repo to build :sweat_smile:), what benefits does the tool have? I can't find anything using it by just grepping through nixpkgs, and it's github repo mainly looks to be shell oriented.

> I use i3 but I can't understand anything about the gif beause it's just too small to read. How do you run the program ? I've tried

Ah, that seems to be because `mpv` doesn't start an external window to swallow without an argument, since just prints some basic usage in the current terminal. When I just run `swallow mpv` I get the same error, but with `swallow mpv /mnt/hdd/movies/...` it works as expected

--- a/pkgs/applications/window-managers/i3/swallow.nix
+++ b/pkgs/applications/window-managers/i3/swallow.nix
@@ -1,0 +1,44 @@
+{ lib
+, buildPythonApplication
+, fetchFromGitHub
+, poetry-core
+, i3ipc
+, xlib
+, six
+}:
+
+buildPythonApplication rec {
+  pname = "i3-swallow";
+  version = "unstable-2022-02-19";
+
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "jamesofarrell";
+    repo = "i3-swallow";
+    rev = "6fbc04645c483fe733de56b56743e453693d4c78";
+    sha256 = "1l3x8mixwq4n0lnyp0wz5vijgnypamq6lqjazcd2ywl2jv8d6fif";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    i3ipc
+    xlib
+    six
+  ];
+
+  # No tests available
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/jamesofarrell/i3-swallow";
+    description = "Swallow a terminal window in i3wm";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    mainProgram = "swallow";
+    maintainers = [ maintainers.ivar ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -26442,6 +26442,8 @@ with pkgs;
 
   i3-resurrect = python3Packages.callPackage ../applications/window-managers/i3/i3-resurrect.nix { };
 
+  i3-swallow = python3Packages.callPackage ../applications/window-managers/i3/swallow.nix { };
+
   i3blocks = callPackage ../applications/window-managers/i3/blocks.nix { };
 
   i3blocks-gaps = callPackage ../applications/window-managers/i3/blocks-gaps.nix { };

